components/raftstore-v2/src/raft/peer.rs

```diff
@@ -210,6 +211,43 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
         Ok(peer)
     }
 
+    pub fn maybe_gen_approximate_buckets<T>(&self, ctx: &StoreContext<EK, ER, T>) {
+        if ctx.coprocessor_host.cfg.enable_region_bucket && !self.region().get_peers().is_empty() {
+            if let Err(e) = ctx
+                .schedulers
+                .split_check
+                .schedule(SplitCheckTask::ApproximateBuckets(self.region().clone()))
+            {
+                error!(
+                    self.logger,
+                    "failed to schedule check approximate buckets";
+                    "err" => %e,
+                );
+            }
+        }
+    }
+
+    #[inline]
+    pub fn region_buckets_mut(&mut self) -> &mut BucketStat {
+        self.region_buckets.as_mut().unwrap()
+    }
+
+    #[inline]
+    pub fn region_buckets(&self) -> &Option<BucketStat> {
+        &self.region_buckets
+    }
+
+    #[inline]
+    pub fn set_region_buckets(&mut self, buckets: BucketStat) {
```

Comment:
Merge this with `reset_region_buckets` into `set_region_buckets(&mut self, buckets: Option<BucketStat>)`

Reply:
yes , I will remove `reset_region_buckets`.